qt/scientific_interfaces/Muon/ALCDataLoadingPresenter.cpp

```diff
@@ -31,37 +33,96 @@ using namespace MantidQt::API;
 
 void ALCDataLoadingPresenter::initialize() {
   m_view->initialize();
 
   connect(m_view, SIGNAL(loadRequested()), SLOT(handleLoadRequested()));
-  connect(m_view, SIGNAL(runsSelected()), SLOT(updateAvailableInfo()));
-  connect(m_view, SIGNAL(runAutoChecked()), SLOT(updateAutoRun()));
-  connect(m_view, SIGNAL(runAutoUnchecked()), SLOT(resetAutoRun()));
+  connect(m_view, SIGNAL(instrumentChangedSignal(std::string)),
+          SLOT(handleInstrumentChanged(std::string)));
+  connect(m_view, SIGNAL(runsChangedSignal()), SLOT(handleRunsChanged()));
+  connect(m_view, SIGNAL(manageDirectoriesClicked()),
+          SLOT(handleManageDirectories()));
+  connect(m_view, SIGNAL(runsFoundSignal()), SLOT(handleRunsFound()));
+  connect(m_view, SIGNAL(autoAddToggledSignal(bool)),
+          SLOT(startWatching(bool)));
+  connect(&m_watcher, SIGNAL(directoryChanged(const QString &)),
+          SLOT(updateDirectoryChangedFlag(const QString &)));
+}
+
+void ALCDataLoadingPresenter::handleRunsChanged() {
+  // Make sure everything is reset
+  m_view->enableLoad(false);
+  m_view->setPath(std::string{});
+  m_view->enableRunsAutoAdd(false);
+  m_view->setAvailableInfoToEmpty();
+  m_view->setLoadStatus("Attempting to find files");
+}
+
+void ALCDataLoadingPresenter::handleRunsFound() {
+  // Do a quick check for an empty input, do nothing in this case
+  if (m_view->getRunsText().empty()) {
+    m_view->setLoadStatus("");
+    return;
+  }
+
+  // Check for errors as files might not have been found
+  if (!m_view->getRunsError().empty()) {
+    m_view->displayError(m_view->getRunsError());
+    m_view->setLoadStatus("");
+    return;
+  }
+
+  // Try update info and enable load
+  try {
+    updateAvailableInfo();
+    m_view->enableLoad(true);
+    m_view->setLoadStatus("Found files");
+  } catch (const std::runtime_error &e) {
+    m_view->displayError(e.what());
+    m_view->setLoadStatus("");
+  }
 }
 
 /**
  * Called when the Load button is clicked.
- * Gets last run, passes it to load method if not "auto".
- * If it was "auto", sets up a watcher to automatically reload on new files.
+ * Displays a warning if trying to load over 200 files
+ * Passes files to load
  */
 void ALCDataLoadingPresenter::handleLoadRequested() {
-  std::vector<std::string> files(m_view->getRuns());
+  auto files = m_view->getFiles();
 
-  // Check not empty
+  // Check there are files
   if (files.empty()) {
-    m_view->displayError(
-        "Error message:\n" + m_view->getRunsErrorMessage() +
-        "\n\nCan specify a list of runs by a dash \ne.g. 1-10\nCan specify "
-        "specific runs with a comma separated list \ne.g. 1-10, 15, 20-30\n "
-        "Range must go in increasing order, \ne.g. 1-10, 15, 20-30");
+    m_view->displayError("The list of files to load is empty");
     return;
   }
 
-  // Now perform the load
-  load(files);
+  // Warning message if trying to load excessive number of files
+  if (files.size() > 200) {
```

Comment:
lets make 200 a const variable at the top of the file. Like this: https://github.com/mantidproject/mantid/blob/master/Framework/Muon/src/CalMuonDetectorPhases.cpp#L25